pandas/core/indexes/multi.py

```diff
@@ -990,15 +990,11 @@ def _constructor(self):
     def _shallow_copy(self, values=None, **kwargs):
         if values is not None:
             names = kwargs.pop("names", kwargs.pop("name", self.names))
```

Comment:
do we need to worry about both name and names being passed?

Reply:
``MultiIndex.from_tuples`` only accepts ``names``, and Mi uses ``name`` in some locations, so yes (but it a part of the ickyness of MultiIndex, so should ideally be refactored away).